Issue/6150

paulp opened the pull request:
Update to #1143.

Commit:
Fixes SI-6150.

Removes the `VectorReusableCBF` and pattern matching on it in optimized `Vector` methods.
Instead, we now have a new `ReusableCBF` instance in `IndexedSeq` and check for equality
when trying to optimize `:+`, `+:` and `updated`.
This overridden `ReusableCBF` is used by `IndexedSeq`, `immutable.IndexedSeq` and
`immutable.Vector`.

The net effect is that calling `:+` and similar methods on a `Vector` instance with
a `CBF` that came from `IndexedSeq` or somewhere lower in the hierarchy will always
create a `Vector` using the optimized method.

paulp committed:
Modified SI-6150 fix to use intended ReusableCBF.

I also realized it didn't have to be lazy, and made it so.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/220/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/926/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/926/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/220/

paulp commented:
Review by @axel22.

paulp committed:
Modified SI-6150 fix again.

Have to keep a sharp eye on those ReusableCBFs.
Now all the indexed sequences should be using the
same instance.

paulp commented:
PLS REBUILD ALL

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/235/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/942/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/942/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/235/
